### PR TITLE
Added rewrite configuration to app gw

### PR DESCRIPTION
**Pull request by djackson-ans**
# Description

- Added the ability to add rewrite rule sets to application gateway
- Fixed the issues with long redirect names and long upm names. 

## Type of change

Please select the option that is relevant.

  - [X] New feature (non-breaking change which adds functionality)

# Checklist:

  - [X] My code follows the style guidelines of this project
  - [X] I have performed a self-review of my code
  - [X] I have commented my code, particularly in hard-to-understand areas
  - [X] I have made corresponding changes to the documentation
  - [X] My changes generate no new warnings e.g. code analysis tooling or general use of the changed code

### CODE REVIEW

File: azure/app-service/examples/advanced/.terraform.lock.hcl [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# This file is maintained automatically by "terraform init".
```
Comment (jk-ans) [inline azure/app-service/examples/advanced/.terraform.lock.hcl:1]: remove .terraform.lock.hcl

**Review by jk-ans — COMMENTED** [review]
Looks good!  Great logic!
Only thing I'd add would be validation but you've already got a TODO for that.

File: azure/app-service/examples/advanced/.terraform.lock.hcl (hunk 1 above)
Comment (djackson-ans) [reply azure/app-service/examples/advanced/.terraform.lock.hcl]: Good spot, that's for a completely different module!

**Review by jk-ans — APPROVED** [review]
LGTM

**Pull request merged**